File: Java-DSA-Ejemplos/algorithms/sorting/InsertionSort.java
package algorithms.sorting;

import java.util.Arrays;

public class InsertionSort {

    public static void insertionSort(int[] arr) {
        int n = arr.length;
        for (int i = 1; i < n; i++) {
            int key = arr[i];
            int j = i - 1;

            // Move elements of arr[0..i-1], that are greater than key,
            // to one position ahead of their current position
            while (j >= 0 && arr[j] > key) {
                arr[j + 1] = arr[j];
                j = j - 1;
            }
            arr[j + 1] = key;
        }
    }

    public static void main(String[] args) {
        int[] myList = {12, 11, 13, 5, 6};

        System.out.println("Lista desordenada: " + Arrays.toString(myList));

        insertionSort(myList);

        System.out.println("Lista ordenada:    " + Arrays.toString(myList));

        int[] anotherList = {64, 34, 25, 12, 22, 11, 90};
        System.out.println("\nLista desordenada: " + Arrays.toString(anotherList));
        insertionSort(anotherList);
        System.out.println("Lista ordenada:    " + Arrays.toString(anotherList));
    }
}

File: Go-DSA-Ejemplos/algorithms/sorting/insertion_sort.go
package sorting

import "fmt"

func InsertionSort(arr []int) {
	n := len(arr)
	for i := 1; i < n; i++ {
		key := arr[i]
		j := i - 1

		for j >= 0 && arr[j] > key {
			arr[j+1] = arr[j]
			j = j - 1
		}
		arr[j+1] = key
	}
}

func main() {
	myList := []int{12, 11, 13, 5, 6}

	fmt.Println("Lista desordenada:", myList)

	InsertionSort(myList)

	fmt.Println("Lista ordenada:   ", myList)

	anotherList := []int{64, 34, 25, 12, 22, 11, 90}
	fmt.Println("\nLista desordenada:", anotherList)
	InsertionSort(anotherList)
	fmt.Println("Lista ordenada:   ", anotherList)
}
